contrib/babelfishpg_common/sql/geography.sql

```diff
@@ -57,16 +57,106 @@ CREATE CAST (sys.GEOGRAPHY AS sys.GEOGRAPHY) WITH FUNCTION sys.GEOGRAPHY(sys.GEO
+									4609, 4610, 4611, 4612, 4613, 4614, 4615, 4616, 4617, 4618, 4619, 4620, 4621, 4622, 4623, 4624, 4625, 4626, 4627, 
+									4628, 4629, 4630, 4632, 4633, 4636, 4637, 4638, 4639, 4640, 4641, 4642, 4643, 4644, 4646, 4657, 4658, 4659, 4660, 
+									4661, 4662, 4663, 4664, 4665, 4666, 4667, 4668, 4669, 4670, 4671, 4672, 4673, 4674, 4675, 4676, 4677, 4678, 4679, 
+									4680, 4682, 4683, 4684, 4686, 4687, 4688, 4689, 4690, 4691, 4692, 4693, 4694, 4695, 4696, 4697, 4698, 4699, 4700, 
+									4701, 4702, 4703, 4704, 4705, 4706, 4707, 4708, 4709, 4710, 4711, 4712, 4713, 4714, 4715, 4716, 4717, 4718, 4719, 
+									4720, 4721, 4722, 4723, 4724, 4725, 4726, 4727, 4728, 4729, 4730, 4732, 4733, 4734, 4735, 4736, 4737, 4738, 4739, 
+									4740, 4741, 4742, 4743, 4744, 4745, 4746, 4747, 4748, 4749, 4750, 4751, 4752, 4753, 4754, 4755, 4756, 4757, 4758, 
+									4801, 4802, 4803, 4804, 4805, 4806, 4807, 4808, 4809, 4810, 4811, 4813, 4814, 4815, 4816, 4817, 4818, 4820, 4821, 
+									4895, 4898, 4900, 4901, 4902, 4903, 4904, 4907, 4909, 4921, 4923, 4925, 4927, 4929, 4931, 4933, 4935, 4937, 4939, 
+									4941, 4943, 4945, 4947, 4949, 4951, 4953, 4955, 4957, 4959, 4961, 4963, 4965, 4967, 4971, 4973, 4975, 4977, 4979, 
+									4981, 4983, 4985, 4987, 4989, 4991, 4993, 4995, 4997, 4999, 7843, 7844, 104001
+								];
+    BEGIN
+        varBin := $1;
+        len := LENGTH(varBin);
+        IF len >= 22 THEN
+			-- We are preprocessing it by removing 2 constant Geometry Type bytes -> 01 0c (currently for Point Type)
+			-- Then adding 5 Bytes -> 01 (little endianess) + 4 Bytes (Geography Type)
+			srid := (get_byte(varBin, 3) << 24) | (get_byte(varBin, 2) << 16) | (get_byte(varBin, 1) << 8) | get_byte(varBin, 0);
+			lat := varbinaryfloat8(CAST ((substring(varBin from 14 for 1) || substring(varBin from 13 for 1) || substring(varBin from 12 for 1) || substring(varBin from 11 for 1) || substring(varBin from 10 for 1) || substring(varBin from 9 for 1) || substring(varBin from 8 for 1) || substring(varBin from 7 for 1))  AS bbf_varbinary));
+            WHILE byte_position < len LOOP
+				-- Get the coordinate to check if it is NaN
+				input_coord := substring(varBin from byte_position + 1 for 8);
+				IF encode(input_coord, 'hex') = encode(coord_NaN, 'hex') THEN
+					isNaN := 1;
+				END IF;
+				byte_position := byte_position + 8;
+			END LOOP;
+			geomType := substring(varBin from 5 for 2);
+			varBin := substring(varBin from 1 for 4) || substring(varBin from 7);
+			IF srid = ANY(valid_srids) AND isNaN = 0 THEN
+				IF encode(geomType, 'hex') = encode(E'\\x010c', 'hex') THEN
+					IF lat >= -90.0 AND lat <= 90.0 THEN
+						newVarBin := E'\\x0101000020' || varBin;
+					ELSE
+						RAISE EXCEPTION 'Error converting data type varbinary to geography.';
+					END IF;
+				ELSE
+					RAISE EXCEPTION 'Currently only Point Type(without Z and M flags) is Supported in Babelfish';
```

Comment:
Are there any other similar error strings in Babelfish? We should make sure the wording matches other places, so users aren't confused by this error if it's thrown.

Reply:
I didn't found any similar error messages for this type of exception as here all unsupported features apart from 2-D Point will be handled, and we can't throw **'feature not supported'** error message here(as used in other places) for now as for Z and M flags for point type will not make sense to throw 'feature not supported' error message as point is supported just Z and M flags are not.